tests/unit/test_index_contents_parse.py

```diff
@@ -0,0 +1,426 @@
+    ]
+
+
+@pytest.mark.parametrize(
+    "content, expected_message_contents",
+    _test__get_contents_parsed_items_invalid_parameters(),
+)
+def test__get_contents_parsed_items_invalid(
+    content: str, expected_message_contents: tuple[str, ...]
+):
+    """
+    arrange: given the index file contents which are invalid
+    act: when get_contents_list_items is called with the index file
+    assert: then InputError is raised with the expected contents in the message.
+    """
+    index_file = types_.IndexFile(title="title 1", content=content)
+
+    with pytest.raises(exceptions.InputError) as exc_info:
+        tuple(index._get_contents_parsed_items(index_file=index_file))
+
+    assert_substrings_in_string(
+        chain(
+            expected_message_contents,
+            "invalid",
+            "item",
+            "contents",
+            "index",
+            index.DOCUMENTATION_INDEX_FILENAME,
+        ),
+        str(exc_info.value).lower(),
+    )
+
+
+def _test__get_contents_parsed_items_parameters():
+    """Generate parameters for the test__get_contents_parsed_items test.
+
+    Returns:
+        The tests.
+    """
+    return [
+        pytest.param(
+            None,
+            (),
+            id="missing file",
+        ),
+        pytest.param(
+            "",
+            (),
+            id="empty file",
+        ),
+        pytest.param(
+            f"""# Contents
+- [{(title_1 := 'title 1')}]({(value_1 := 'value 1')})""",
+            (
+                factories.IndexParsedListItemFactory(
+                    whitespace_count=0, reference_title=title_1, reference_value=value_1, rank=0
+                ),
+            ),
+            id="single item",
+        ),
+        pytest.param(
+            f"""# Contents
+-  [{(title_1 := 'title 1')}]({(value_1 := 'value 1')})""",
+            (
+                factories.IndexParsedListItemFactory(
+                    whitespace_count=0, reference_title=title_1, reference_value=value_1, rank=0
+                ),
+            ),
+            id="single item multiple whitespace after leader",
+        ),
+        pytest.param(
+            f"""# Contents
+- [{(title_1 := 'title 1')}]({(value_1 := 'value 1')}) """,
+            (
+                factories.IndexParsedListItemFactory(
+                    whitespace_count=0, reference_title=title_1, reference_value=value_1, rank=0
+                ),
+            ),
+            id="single item trailing whitespace",
+        ),
+        pytest.param(
+            f"""# Contents
+- [{(title_1 := 'title 1')}]({(value_1 := 'value 1')})  """,
+            (
+                factories.IndexParsedListItemFactory(
+                    whitespace_count=0, reference_title=title_1, reference_value=value_1, rank=0
+                ),
+            ),
+            id="single item multiple trailing whitespace",
+        ),
+        pytest.param(
+            f"""# Contents
+- [{(title_1 := 'title 1')}]({(value_1 := 'value 1')})""",
```

Comment:
Shouldn't it be:
```
f"""# Contents


- [{(title_1 := 'title 1')}]({(value_1 := 'value 1')})""",
```

Reply:
yes good catch, should be fixed on next push